lib/fastlane/actions/verify_build.rb

```diff
@@ -0,0 +1,172 @@
+# rubocop:disable Style/ConditionalAssignment
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/AbcSize
+
+require 'plist'
+require 'terminal-table'
+
+module Fastlane
+  module Actions
+    module SharedValues
+      VERIFY_BUILD_CUSTOM_VALUE = :VERIFY_BUILD_CUSTOM_VALUE
+    end
+    class VerifyBuildAction < Action
+      def self.run(params)
+        Dir.mktmpdir do |dir|
+          # prepare
+          values = {}
+
+          ipa_path = Actions.lane_context[SharedValues::IPA_OUTPUT_PATH] || ''
+          raise "Unable to find ipa file '#{ipa_path}'." unless File.exist?(ipa_path)
+          ipa_path = File.expand_path(ipa_path)
+
+          # unzip
+          `unzip '#{ipa_path}' -d #{dir}`
+          raise "Unable to unzip ipa".red unless $? == 0
+
+          app_path = File.expand_path("#{dir}/Payload/*.app")
+
+          # extract
+          `cd #{dir} && codesign -d --extract-certificates #{app_path} 2>/dev/null`
```

Comment:
It looks like you would be able to get the data you want from one command with codesign. Running `codesign -vv -d {path_to_app_bundle}` gives me provisioning type and team identifier. Do you think this is sufficient? It may require some changes to your parsing code though.